src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,142 @@
 package core.basesyntax;
 
+import static java.lang.Math.abs;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private final int DEFAULT_INITIAL_CAPACITY = 16;
+    private final double DEFAULT_LOAD_FACTOR = 0.75;
+    private HashObject<K, V>[] hashMap;
+    private int capacity = DEFAULT_INITIAL_CAPACITY;
+    private int size = 0;
+
+    public MyHashMap() {
+        hashMap = new HashObject[DEFAULT_INITIAL_CAPACITY];
+    }
+
+    public class HashObject<K, V> {
+        private K key;
+        private V value;
+        private HashObject<K, V> next;
+
+        public HashObject(K key, V value, HashObject<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public void enlarge() {
```

Comment:
The enlarge() method could be implemented more efficiently. Currently, it uses keySet() and getValue() in a loop, which results in multiple full iterations over the map. Consider iterating over the map's entries just once, storing them in a temporary collection, and then putting them back into the resized map.